File: frontend/hc_professional/src/api_types/Web_AdherenceMonthResponse.ts
export interface WebAdherenceMonthResponse {
    adherence_days: WebAdherenceDayEntry[];
    month:          number;
    month_pdc:      number;
    pdc_target:     number;
    year:           number;
}

export interface WebAdherenceDayEntry {
    date:       Date;
    id:         number;
    status:     AdherenceStatusEnum;
    symptoms:   string[];
    video_link: null | string;
}

export enum AdherenceStatusEnum {
    AppRecorded = "app_recorded",
    ProviderReconciled = "provider_reconciled",
    TechnicalMiss = "technical_miss",
    UnverifiedAbsence = "unverified_absence",
}

File: frontend/patient/app/src/main/java/com/pinghtdog/amping/api_schemas/Web_AdherenceMonthResponse.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json                      = Json { allowStructuredMapKeys = true }
// val webAdherenceMonthResponse = json.parse(WebAdherenceMonthResponse.serializer(), jsonString)

package com.pinghtdog.amping.api_schemas

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class WebAdherenceMonthResponse (
    @SerialName("adherence_days")
    val adherenceDays: List<WebAdherenceDayEntry>,

    val month: Long,
    val year: Long
)

@Serializable
data class WebAdherenceDayEntry (
    @SerialName("adherence_type")
    val adherenceType: String,

    val date: String,
    val symptoms: List<String>,

    @SerialName("video_link")
    val videoLink: String? = null
)
